composer/algorithms/distillation/distillation.py

```diff
@@ -0,0 +1,193 @@
+            elif isinstance(base_loss, torch.Tensor) or isinstance(base_loss, float):
+                state.loss = self.org_loss_weight * base_loss + self.kd_loss_weight * kd_loss
+
+    def _move_teacher_model_to_device(self, teacher_model: Union[ComposerModel, nn.Module, torch.Tensor],
+                                      student_model: Union[ComposerModel, nn.Module], device: Device):
+        """Moves the teacher model to student model device.
+
+        Args:
+            teacher_model (Union[ComposerModel, nn.Module, torch.Tensor]): Teacher model.
+            student_model (ComposerModel): Student model.
+            device (Device): Device to move teacher model to.
+        """
+        device = get_device(device)
+        with torch.no_grad():
+            # check if model is already on tpu
+            if isinstance(device, DeviceTPU) and 'xla' not in str(next(student_model.parameters()).device):
+                raise ValueError(
+                    'Use model.to(xm.xla_device()) to set the model to the TPU before providing to the trainer.')
+            else:
+                assert isinstance(teacher_model, nn.Module)
+                device.module_to_device(teacher_model)
+
+
+class KLDivergence(nn.Module):
+    """KL Divergence loss for distillation.
+
+    Args:
+        temperature (float, optional): Temperature for softmax. The temperature controls the entropy
+            of the outputted probability distribution. As ``temperature`` approaches
+            ``0``, the probability for the largest element approaches 1, and as
+            ``temperature`` approaches ``inf``, the probability for each element will be
+            the same. Default: ``4.0``.
+        softmax_dim (int, optional): The dimension across which to compute the softmax.
+            Tradition—and all theory and common sense—dictate that the only correct
+            dimension across which to compute the softmax is the class dimension
+            (``-1`` here). However, we found (by accident) that we obtained superior
+            results when computing the softmax across tokens in a sequence (dimension =
+            ``0``) for masked language modeling (MLM) pretraining. Accordingly, users may
+            want to set ``softmax_dim=0`` when using distillation during MLM pretraining.
+            Default: ``-1``.
+
+    """
+
+    def __init__(
+        self,
+        temperature: float = 4.0,
+        softmax_dim: int = -1,
+    ):
+        super(KLDivergence, self).__init__()
+        self.temperature = temperature
+        self.softmax_dim = softmax_dim
+
+    def forward(self, y_student: torch.Tensor, y_teacher: torch.Tensor):
+        p_student = F.log_softmax(y_student / self.temperature, dim=self.softmax_dim)
+        p_teacher = F.softmax(y_teacher / self.T, dim=self.softmax_dim)
+        loss = F.kl_div(p_student, p_teacher, size_average=False) * (self.temperature**2) / y_student.shape[0]
```

Comment:
I know this is my code, but I think size_average is being deprecated in the near future

I think 

```suggestion
        loss = F.kl_div(p_student, p_teacher, reduction='batchmean') * (self.temperature**2) 
```

should be equivalent.